File: app-api-flutter/contato.js
module.exports = (app) => {
   var array = [
      {'id':'1', 'nome':'Jaqson', 'email':'jaqson@upf.br', 'telefone':'54 65664-0536'},
      {'id':'2', 'nome':'Fulano', 'email':'fulano@upf.br', 'telefone':'54 87878-0636'}
   ];

   app.get('/contato', (req, res) => {
      res.json(array);
   });

   app.post('/contato', (req, res, next) => {
      if (req.body.id === null){
         req.body.id = (array.length+1).toString();
      }
      array.push(req.body);
      res.json({success: "Incluído"});
   });
 
   app.put('/contato', (req, res) => {
      for (i = 0; i < array.length; i++) {
         if (array[i].id === req.body.id){
            array[i] = req.body;
            return res.json({success: "Alterado com sucesso."});
         }
      };
      return res.json({aviso: "Nada alterado."});
   });

   app.delete('/contato/:id', (req, res) => {
      for (i = 0; i < array.length; i++) {
         if (array[i].id === req.params.id){
            array.splice(i, 1);
            return res.json({success: "Excluído com sucesso."});
         }
      };
      return res.json({aviso: "Nada excluído."});
   }); 

   app.get('/contato/:id', (req, res) => {
      for (i = 0; i < array.length; i++) {
         if (array[i].id === req.params.id){
            return res.json(array[i]);
         }
      };
      return res.json({aviso: "Não encontrado."});
   });


 }

File: lib/model/contato.dart
class Contato {
  String? id;
  String? nome;
  String? email;
  String? telefone;

  Contato() {
    id = null;
    nome = "";
    email = "";
    telefone = "";
  }

  Contato.init(this.id, this.nome, this.email, this.telefone);

  factory Contato.fromMap(Map<String, dynamic> json) {
    return Contato.init(
        json['id'], json['nome'], json['email'], json['telefone']);
  }

  Map<String, dynamic> toMap() =>
      {'id': id, 'nome': nome, 'email': email, 'telefone': telefone};

  @override
  toString() {
    return '$nome - $telefone';
  }

  bool operator ==(other) {
    return (other is Contato && other.id == id);
  }
}
